crates/types/src/config/common.rs

```diff
@@ -574,11 +576,64 @@ pub enum MetadataStoreClient {
     },
 }
 
+#[derive(Debug, serde::Deserialize)]
+#[serde(
+    tag = "type",
+    rename_all = "kebab-case",
+    rename_all_fields = "kebab-case"
+)]
+enum MetadataStoreClientShadow {
```

Comment:
Maybe add a comment explaining when we can remove this enum (once we no longer support the `address` case).

Reply:
Done :) will merge after CI passes.